lib/logstash/inputs/udp.rb

```diff
@@ -147,8 +150,12 @@ def inputworker(number, codec)
         break if payload == :END
 
         ip_address = client[3]
-
-        codec.decode(payload) { |event| push_decoded_event(ip_address, event) }
+        if @metadata
+          metadata_to_codec = {}
+          metadata_to_codec["port"] = client[1]
+          metadata_to_codec["host"] = client[3]
+        end
```

Comment:
this could be created with a single assignment:
```
metadata_to_codec = { "port" => client[1], "host" => client[3] }
```

Reply:
Thanks for the hint, I'm happy to change this.